### PR TITLE
feat(insights): funnel on sessions

### PR DESCRIPTION
**Pull request by mariusandra**
## Problem

People would like to aggregate their funnels based on `properties.$session_id`

## Changes

Now they can:

I introduced a new filter for funnels called `funnel_aggregate_by_hogql`. This property can accept any HogQL and will use it instead of `person_id` for funnel aggregation. 

The only gotcha: person properties are not supported. It seemed like a lot of hassle to get those to work nicely, so I opted to hide them for now. To make the UI simple and predictable, instead of adding a new "custom hogql" field, I just added a new option into the aggregation dropdown called "unique sessions". That should do it for now to solve a customer requests. Once funnels are rewritten on top of HogQL, we can expose this to users as an additional bonus.

I also disabled the person modal for these custom properties.

## How did you test this code?

Wrote a test to cover the new missing cases. Clicked around in the interface with both data exploration on and off.

### CODE REVIEW

File: frontend/src/scenes/funnels/FunnelBarChart/FunnelBarChart.tsx [hunk 1]
```diff
@@ -15,19 +15,27 @@ import { funnelDataLogic } from '../funnelDataLogic'
 
 export function FunnelBarChartDataExploration(props: ChartParams): JSX.Element {
     const { insightProps } = useValues(insightLogic)
+    const { canOpenPersonModal } = useValues(funnelLogic)
```
Comment (thmsobrmlr) [inline frontend/src/scenes/funnels/FunnelBarChart/FunnelBarChart.tsx:18]: This should be implemented in `funnelDataLogic` and based on the query.

**Review by thmsobrmlr — CHANGES_REQUESTED** [review]
Nice! People have been asking for this a lot lately.

Some of the things should be based on the query for data exploration to work once we stop converting queries to filters. Funnels are already fully converted and thus work without it (but other visualizations are not, which is why we still have the conversion).

You can disable the conversion by commenting out [line 223 in insightVizDataLogic](https://github.com/PostHog/posthog/blob/b74dd435f83bfc3db77bccbc8c1d35f693dbffd4/frontend/src/scenes/insights/insightVizDataLogic.ts#L223).

Basically what this needs is:
- Implementing `canOpenPersonModal` in `funnelDataLogic` as well and using it in the data exploration variants
- Implementing the changes to `aggregationTargetLabel` in `funnelDataLogic` as well

The person modal isn't deactivated when clicking on the bar chart as well. I think this is because we replace the openPersonsModalForSeries with a no-op where we deactivate it right now. Should probably be based in `canOpenPersonModal` as well.

**Comment by mariusandra** [comment]
This could use a re-review

**Review by thmsobrmlr — APPROVED** [review]
Works like a charm now.
